web/src/components/prompt.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from 'react';
+import { Prompt as ReactRouterPrompt } from 'react-router-dom';
+
+interface PromptProps {
+  when: boolean;
+  message: string;
+}
+
+/**
+ * Wrapper around react-router Prompt adding confirmation on tab close and external navigation.
+ */
+export const Prompt: React.FC<PromptProps> = (props) => {
+  const { when } = props;
+  useEffect(() => {
+    if (!when) {
+      return () => null;
+    }
+
+    function listener(event: BeforeUnloadEvent) {
+      // Cancel the event as stated by the standard.
+      event.preventDefault();
+      // Chrome requires returnValue to be set.
+      event.returnValue = '';
+    }
+
+    window.addEventListener('beforeunload', listener);
+    return () => window.removeEventListener('beforeunload', listener);
+  }, [when]);
+
+  return (
+    <ReactRouterPrompt {...props}/>
```

Comment:
As far as I can see we could directly use `Prompt` from `react-router-dom` in the components, as all the event listener set up is already handled by it. Was there a specific reason why you had to wrap it with this component?

Reply:
I'm not sure I follow - the `Prompt` from `react-router-dom` does not handle `beforeunload` (https://github.com/ReactTraining/react-router/pull/4372 was never merged as far as I can see). I did tested and no warning was issued when simply closing the tab.

Reply:
Right, I should not review PRs before my second coffee kicks in. Thanks for the elaboration!